src/Logary.Tests/Metrics.fs

```diff
@@ -175,4 +180,14 @@ let tests = [
   testCase "histogram get exponentialBuckets" <| fun () ->
     let histogramConf = HistogramConf.create("some histogram", "some histogram") |> HistogramConf.exponentialBuckets 5. 5. 5
     histogramConf.buckets |> Expect.sequenceEqual "should have buckets" [ 5.; 25.; 125.; 625.; 3125.; ]
+
+  testCase "avoid high cardinality" <| fun () ->
+    let registry = new MetricRegistry()
+    let gauge = {basicConf with labelNames = [| "user_id" |]} |> GaugeConf.create |> registry.registerMetric
+    for userId in 1..150 do
+      gauge.labels [| string userId |] |> ignore
+
+    Expect.throws "should throw if reach the high cardinality limit" (fun () ->
```

Comment:
Can we make it non-throwable by default, perhaps? That way we never throw exceptions, and we have to "opt in" to throwing them. So by default, we should perhaps just warn if we have a high cardinality data set (with internal loggar at warn) and then throw if the user opts into that.

Reply:
how about this ?

Reply:
non-throwable by default, needs logger to be passed in, do you have any ideas how to make it default since metric can be created outside of logary's registry (which have its runtime info with internal logger).

Reply:
👍